R/util.R

```diff
@@ -16,18 +16,36 @@ read_file <- function(path) {
 ## Same logic as jsonvalidate, possibly overkill here. Perhaps we
 ## always want a path to the input?
 read_input <- function(x, name = deparse(substitute(x))) {
+  if (is.null(x)) {
+    ## This might need to be tuneable - for the initial load we need a
+    ## null at least.
+    return(jsonlite::unbox("null"))
+  }
   if (read_input_is_filename(x)) {
     if (!file.exists(x)) {
       stop(sprintf("'%s' looks like a filename but does not exist", name),
            call. = FALSE)
     }
     x <- read_file(x)
   }
-  V8::JS(x)
```

Comment:
we no longer pass in raw unvalidated JSON as this lead to nasty bugs and was not compatible with the v8/R callback mechanism. Now we pass in a string and use `JSON.parse` to unpack it in JS